src/components/flyout/flyout.tsx

```diff
@@ -340,57 +339,48 @@ export const EuiFlyout = forwardRef(
       );
     }
 
-    const flyoutContent = (
-      <Element
-        {...(rest as ComponentPropsWithRef<T>)}
-        role={role}
-        className={classes}
-        tabIndex={-1}
-        style={newStyle || style}
-        ref={setRef}
-      >
-        {closeButton}
-        {children}
-      </Element>
-    );
-
+    const isDefaultConfiguration = ownFocus && !isPushed;
+    const onClickOutside =
+      (isDefaultConfiguration && outsideClickCloses !== false) ||
+      outsideClickCloses === true
+        ? onClose
+        : undefined;
```

Comment:
Also super optional feedback, but this is pretty confusing to read - I wonder if a useMemo with early returns would be easier to parse?

```suggestion
    const onClickOutside = useMemo(() => {
      if (outsideClickCloses === false) return undefined;
      if (outsideClickCloses === true) return onClose;
      if (isDefaultConfiguration) return onClose;
      return undefined;
    }, [outsideClickCloses, isDefaultConfiguration]);
```

After rereading again I'm not super sure this is a huge improvement so feel free to disregard this lol